deploy/setup.sh

```diff
@@ -0,0 +1,280 @@
+    fi
+
+    # for ubuntu 15
+    if $(lsb_release -d | grep -q "15"); then
+        echo "deb https://apt.dockerproject.org/repo ubuntu-wily main" | tee /etc/apt/sources.list.d/docker.list
+    fi
+
+    # for linux mint
+    if $(lsb_release -d | grep -q "Mint"); then
+        echo "deb https://apt.dockerproject.org/repo ubuntu-trusty main" | tee /etc/apt/sources.list.d/docker.list
+    fi
+
+    apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
+}
+
+get_dependency_software() {
+    echo "updating apt-get......"
+    result=$(sudo apt-get update)
+    if grep -q "Could not resolve" <<< $result; then
+        echo "Could not update apt-get, please solve it"
+        exit
+    fi
+    echo "installing git python-setuptools python-dev python-pip"
+    result=$(sudo apt-get install -y git python-setuptools python-dev python-pip)
+    if grep -q "Unable to lacate" <<< $result; then
+        echo "Could not install dependancy software, pls install docker manually"
+        exit
+    fi
+    echo "installing autoconf libtool tomcat7 "
+    result=$(sudo apt-get install -y autoconf automake libtool tomcat7 )
+    if grep -q "Unable to lacate" <<< $result; then
+        echo "Could not install dependancy software, pls install docker manually"
+        exit
+    fi
+}
+
+set_envirement() {
+    echo "git clone source code from Github, and solve python dependency"
+    cd /home &&  rm -rf opentech
+    mkdir opentech && cd opentech
+    rm -r open-hackathon
+    echo "git cloning open-hackathon source code"
+    result=$(sudo git clone https://github.com/YaningX/open-hackathon.git)
+    if grep -q "unable to access" <<< $result; then
+        echo "Could not git clone open-hackathon source code, pls check your network"
+        exit
+    fi
+    cd /home/opentech/open-hackathon
+    echo "pip is installing required python library"
+    result=$(sudo pip install -r open-hackathon-server/requirement.txt)
+
+    result=$(sudo pip install -r open-hackathon-client/requirement.txt)
+    cp open-hackathon-server/src/hackathon/config_sample.py open-hackathon-server/src/hackathon/config.py
+    cp open-hackathon-server/src/hackathon/config_sample.py open-hackathon-server/src/hackathon/config.py
+}
+
+install_mongodb() {
+   result=$(sudo service mongod status)
+   if ! $(grep "unrecognized service" <<< $result); then
+       echo "mongodb is installed"
+       return
+   fi
+   sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 7F0CEB10
+   if $(lsb_release -d | grep -q "14"); then
+        echo "deb http://repo.mongodb.com/apt/ubuntu trusty/mongodb-enterprise/3.0 multiverse" | tee /etc/apt/sources.list.d/mongodb-enterprise.list
+   fi
+   if $(lsb_release -d | grep -q "Mint"); then
+        echo "deb http://repo.mongodb.com/apt/ubuntu trusty/mongodb-enterprise/3.0 multiverse" | tee /etc/apt/sources.list.d/mongodb-enterprise.list
```

Comment:
If my memory doesn't cheat me, `mongodb enterprise` includes a `commercial license`. Please go and check it. If it is, we must install `mongodb community` version.